linux/devdoc/error_handling_requirements.md

```diff
@@ -0,0 +1,41 @@
+# error handling
+
+## Overview
+
+Linux implentaton of GetLastError and SetLastError 
+
+## Exposed API
+
+```c
+uint64_t last_error_code;
+
+MOCKABLE_FUNCTION(, void, error_handling_set_last_error, uint32_t, err_code);
+MOCKABLE_FUNCTION(, uint32_t, error_handling_get_last_error);
+```
+
+### set_last_error
+
+```c
+MOCKABLE_FUNCTION(, void, error_handling_set_last_error, uint32_t, err_code);
+```
+`error_handling_set_last_error` implements the PAL wrapper for the SetLastError function from Windows.
+
+`error_handling_set_last_error` shall succeed and assign a non-`NULL` value to `last_error_code`.
+
+`error_handling_set_last_error` shall call `interlocked_exchange_64` with `err_code` and `last_error_code`.
+
+
+If the `err_code` is `NULL`, `error_handling_set_last_error` shall not change `last_error_code`.
+
+### get_last_error
+
+```c
+MOCKABLE_FUNCTION(, uint32_t, error_handling_get_last_error);
+```
+`error_handling_get_last_error` implements the PAL wrapper for the GetLastError function from Windows.
+
+On success, `error_handling_get_last_error` shall return a non-`NULL` value.
```

Comment:
Since uint32_t is not a pointer, it can not be NULL.  Probably should say something like 

`error_handling_get_last_error` shall return the value last set through set_last_error or 0 (0 means everything is OK).